matplotcheck/base.py

```diff
@@ -675,50 +678,58 @@ def assert_xydata(
         None :
             Nothing if no legends overlap, otherwise throws ``AssertionError`` with message `m`
         """
+        if xy_expected is None:
+            return
```

Comment:
Can you explain using return vs pass here?

Reply:
The expected behavior is that the assertion passes if `xy_expected` is `None`.

In a place like this, `pass` is effectively an empty line. If it remains unchanged, and `xy_expected` is `None`, then the function continues to try to evaluate `xy_expected` like it contains data, and eventually throws an error.

By replacing it with `return`, the assertion passes as expected.